### ISSUE TITLE
Elm API suggestion (breaking change) – use records for most parameters to `call` and `callNoArgs`

### ISSUE THREAD

**Issue by GabrielBoehme13** [lobanov/elm-taskport#2]
To aid future code readability, I'd like to request the use of records when specifying the function name and JSON decoder/encoder arguments for `call` and `callNoArgs` (since I believe they would always be specified together).

Below are examples of how calls with the current & proposed API would look:

```Elm
-- current API: I don't find this clear enough at a glance when reading the code
testProcedureConfusing : Int -> Task.Task (Error String) Int
testProcedureConfusing =
  TaskPort.call "testProcedure" JD.int JD.string JE.int


-- proposed API: define my test procedure
testProcedure : Int -> Task.Task (Error String) Int
testProcedure =
  TaskPort.call
    { functionName = "testProcedure"
    , argsEncoder = JE.int
    , bodyDecoder = JD.int
    , errorDecoder = JD.string
    }


-- an example task to call the test procedure with an argument
taskForTestProcedureSpecifying3 : Task.Task (Error String) Int
taskForTestProcedureSpecifying3 =
  testProcedure 3
```

Here's one way `call` could be defined to support this:

```Elm
call :
  { functionName : String
  , argsEncoder : args -> JE.Value
  , bodyDecoder : JD.Decoder body
  , errorDecoder : JD.Decoder error
  }
  -> args
  -> Task.Task (Error error) body
call { functionName, bodyDecoder, errorDecoder, argsEncoder } args =
  callWithJson functionName bodyDecoder errorDecoder <| argsEncoder args
```

**Comment by lobanov** [comment]
Thank you @GabrielBoehme13 this is a good suggestion. Admittedly I have not paid much attention to the ergonomics of the public interface, because I had assumed (and intended myself) it is going to be wrapped using constructs specific to the API being exposed. For example, your `testProcedure` is ultmately what is going to be invoked in the application logic like in `taskForTestProcedureSpecifying3`. 

I will keep it open to gather more feedback and see how people use it. I also want to experiment further with error handling and see if there is more convenient way to do this. It's likely that the public interface will evolve in the direction you suggested though.

**Comment by lobanov** [comment]
@GabrielBoehme13 I think there is now a critical mass of new features (#7 #6 #3 in particular, plus #8 and #9 todo) that warrant a breaking change. I also adopted the suggestion here in newly added `callNS` and `callNoArgsNS` functions added as part of #7 and #6. I will be working on the new public API along the lines of what you had suggested, but in the meantime would appreciate any feedback on the above.

**Comment by lobanov** [comment]
Implemented in v2.0.0
